crates/biome_js_analyze/tests/specs/nursery/noBarrelFile/invalid.ts.snap

```diff
@@ -0,0 +1,130 @@
+---
+source: crates/biome_js_analyze/tests/spec_tests.rs
+expression: invalid.ts
+---
+# Input
+```ts
+export * from "foo";
+export * as bar from "foo";
+export { foo } from "foo";
+export { foo, type Bar } from "foo";
+export { baz, qux } from "foobar";
+export { module as module1 } from "./module1";
+export { default as module2 } from "./module2";
+```
+
+# Diagnostics
+```
+invalid.ts:1:1 lint/nursery/noBarrelFile ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
+
+  ! Avoid barrel files, they slow down performance, and cause large module graphs with modules that go unused.
+  
+  > 1 │ export * from "foo";
+      │ ^^^^^^^^^^^^^^^^^^^^
+    2 │ export * as bar from "foo";
+    3 │ export { foo } from "foo";
+  
+  i Check this thorough explanation to better understand the context.
```

Comment:
I think we should emit only one diagnostic in this case. The issue is the whole file, not each export. 

So maybe we should match the whole `JsModule` instead of matching each export

Reply:
Makes sense to me 👍